vars/abapEnvironmentPipelineCleanup.groovy

```diff
@@ -0,0 +1,23 @@
+import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.GenerateStageDocumentation
+import groovy.transform.Field
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field STAGE_STEP_KEYS = []
+@Field Set STEP_CONFIG_KEYS = []
+@Field Set PARAMETER_KEYS = []
+/**
+ * This stage cleans up the ABAP Environment Pipeline run
+ */
+void call(Map parameters = [:]) {
+    def script = checkScript(this, parameters) ?: this
+    def stageName = parameters.stageName?:env.STAGE_NAME
+
+    if(env.BRANCH_NAME == parameters.script.commonPipelineEnvironment.configuration.runStage?.get("Prepare System")) {
+        cloudFoundryDeleteService script: parameters.script
```

Comment:
Maybe we can add a user input here for confirming that the ABAP system has been deprovisioned successfully. Since the service instance incur costs I think this would make sense here.

Reply:
The goal is a fully automated pipeline (at least when running successfully).

Reply:
Ok. I just figured since you were using a user input in line 21 in `vars/abapEnvironmentPipelineStagePrepareSystem.groovy` step that it would also make sense here..

Reply:
The input in `vars/abapEnvironmentPipelineStagePrepareSystem.groovy` is the current workaround. I hope, we can get rid of it in the near future.